addon/services/street.js

```diff
@@ -18,54 +27,76 @@ export default class StreetService extends GWRService {
     return this.cache(street);
   }
 
-  get language() {
-    return languageOptions[this.intl.primaryLocale?.split("-")?.[0]];
+  get primaryLanguage() {
+    return this.intl.primaryLocale?.split("-")?.[0];
   }
 
-  async search(query = {}) {
-    if (this.cachedLanguageOverride) {
-      query.language = this.cachedLanguageOverride;
-    }
+  get language() {
+    return this.languages.languageToCode(this.primaryLanguage);
+  }
 
-    const searchResult = await super.search(query, query.EGID, {
+  @task
+  *_search(query) {
+    return yield super.search(query, query.EGID, {
       xmlMethod: "getStreet",
       urlPath: "streets",
       listModel: StreetList,
       listKey: "street",
       searchKey: "streetWithoutStreetGeometryType",
     });
+  }
 
-    if (!searchResult) {
-      let results = [];
-
-      for (const [lang, code] of Object.entries(languageOptions)) {
-        query.language = code;
-
-        results.push({
-          lang: [lang, code],
-          result: super.search(query, query.EGID, {
-            xmlMethod: "getStreet",
-            urlPath: "streets",
-            listModel: StreetList,
-            listKey: "street",
-            searchKey: "streetWithoutStreetGeometryType",
-          }),
-        });
-      }
-
-      results = await Promise.allSettled(
-        results.map(async ({ lang, result }) => {
-          return { lang, result: await result };
+  resetCachedLanguages() {
+    this.cachedLanguageOverrides = new Set([]);
+  }
+
+  async searchMultiple(query, useCachedLanguagesOnly) {
+    let results = [];
+
+    for (const lang of useCachedLanguagesOnly
+      ? this.cachedLanguageOverrides
+      : this.languages.availableLanguages) {
```

Comment:
Doesn't this mean we will be performing a search over all languages the first time around instead of first trying only the primary language?

Reply:
I guess this is related to the discussion:

> Update after short discussion with @czosel:
> We want to stay very defensive with caching and fire one or two requests more rather then getting trapped in some request cache shenanigans. Opting for only canton relevant languages will most effectively cut down request count.

Reply:
Code looks good to me, needs a rebase :+1: Did we have the chance to test it in a multilingual municipality?

Reply:
* `rebase done` 
* yes, I've tested it in a multilingual municipality, but a second test is welcome